src/scripts/tenantDataMigrations/migrate-existing-tenant-data.js

```diff
@@ -0,0 +1,1965 @@
+					columns: 'tenant_code, report_code',
+				},
+				{ table: 'feedbacks', name: 'idx_feedbacks_citus_tenant_user', columns: 'tenant_code, user_id' },
+
+				// Configuration indexes
+				{ table: 'default_rules', name: 'idx_default_rules_citus_tenant_type', columns: 'tenant_code, type' },
+				{ table: 'question_sets', name: 'idx_question_sets_citus_tenant_code', columns: 'tenant_code, code' },
+				{
+					table: 'role_extensions',
+					name: 'idx_role_extensions_citus_tenant_title',
+					columns: 'tenant_code, title',
+				},
+			]
+
+			let indexesCreated = 0
+			let indexesSkipped = 0
+
+			for (const idx of citusIndexes) {
+				try {
+					// Check if table exists
+					const tableExists = await this.sequelize.query(
+						`SELECT EXISTS (SELECT FROM information_schema.tables WHERE table_name = :tableName)`,
+						{
+							replacements: { tableName: idx.table },
+							type: Sequelize.QueryTypes.SELECT,
+						}
+					)
+
+					if (!tableExists[0].exists) {
+						console.log(`⚠️  Table ${idx.table} does not exist, skipping index`)
+						indexesSkipped++
+						continue
+					}
+
+					// Create index with IF NOT EXISTS to avoid conflicts
+					await this.sequelize.query(
+						`CREATE INDEX IF NOT EXISTS "${idx.name}" ON "${idx.table}" (${idx.columns})`
+					)
+					console.log(`✅ Created Citus index: ${idx.name}`)
+					indexesCreated++
+				} catch (error) {
+					console.log(`⚠️  Could not create index ${idx.name}: ${error.message}`)
+					indexesSkipped++
+				}
+			}
+
+			console.log(`\n📈 Citus Index Creation Summary:`)
+			console.log(`✅ Successfully created: ${indexesCreated} indexes`)
+			console.log(`⚠️  Skipped/Failed: ${indexesSkipped} indexes`)
+			console.log(`📊 Total attempted: ${citusIndexes.length} indexes`)
+
+			if (indexesCreated > 0) {
+				console.log('\n🚀 Citus Performance Tips:')
+				console.log('• All indexes are tenant-first for optimal Citus performance')
+				console.log('• Always include tenant_code in WHERE clauses')
+				console.log('• Cross-tenant queries will be slower (avoid when possible)')
+				console.log('• Monitor performance: SELECT * FROM citus_stat_statements ORDER BY total_time DESC;')
+			}
+
+			console.log('\n💡 Next Steps:')
+			console.log('• Ensure all tables are distributed with: SELECT * FROM citus_tables;')
+			console.log('• Run ANALYZE on all tables to update statistics')
+			console.log('• Test query performance with tenant-aware queries')
+		} catch (error) {
+			console.log(`⚠️  Citus index creation failed: ${error.message}`)
+			console.log('💡 This is expected for non-Citus PostgreSQL installations')
+		}
+	}
+
+	/**
+	 * Create unique indexes after data migration and duplicate cleanup
+	 * NOTE: Should only be run after duplicate data cleanup
+	 */
+	/**
+	 * Fix existing UNIQUE constraints to include tenant_code for proper multi-tenant isolation
+	 * This ensures all unique constraints are tenant-aware
+	 */
+	async fixUniqueConstraints() {
```

Comment:
UNIQUE constraints are schema concerns and must be handled via Sequelize migrations.
Please move all UNIQUE constraint changes into migrations.